radix-engine/src/system/system_callback.rs

```diff
@@ -1125,6 +1127,15 @@ impl<C: SystemCallbackObject> KernelCallbackObject for System<C> {
             None
         };
 
+        let debug_information = match (&costing_module.detailed_cost_breakdown,) {
+            (Some(detailed_cost_breakdown),) => Some(TransactionDebugInformation {
+                detailed_execution_cost_breakdown: detailed_cost_breakdown
```

Comment:
If we create a `costing_module.deconstruct()` which splits out the `detailed_cost_breakdown` from the `fee_reserve` etc then can we avoid the expensive clone here?

Reply:
Sounds good to me. In this case avoiding the clone didn't require adding a new method. The current context already owns the `CostingModule` so I can simply operate over the owned value instead of operating over a reference and this avoids the clone.